pysat/tests/instrument_test_class.py

```diff
@@ -47,17 +49,93 @@ def initialize_test_inst_and_date(inst_dict):
 class InstTestClass(object):
     """Provide standardized tests for pysat instrument libraries.
 
-    Note: Not diretly run by pytest, but inherited through test_instruments.py
+    Note
+    ----
+    Uses class level setup and teardown so that all tests use the same
+    temporary directory. We do not want to geneate a new tempdir for each test,
+    as the load tests need to be the same as the download tests.
+
+    Not directly run by pytest, but inherited through test_instruments.py
+
+    Users will need to run `apply_marks_to_tests` before setting up the test
+    class.
+
     """
 
-    module_attrs = ['platform', 'name', 'tags', 'inst_ids',
-                    'load', 'list_files', 'download']
-    inst_attrs = ['tag', 'inst_id', 'acknowledgements', 'references',
-                  'inst_module']
-    inst_callable = ['load', 'list_files', 'download', 'clean', 'default']
-    attr_types = {'platform': str, 'name': str, 'tags': dict,
-                  'inst_ids': dict, 'tag': str, 'inst_id': str,
-                  'acknowledgements': str, 'references': str}
+    def setup_class(self):
+        """Initialize the testing setup once before all tests are run."""
+
+        # Define standard attributes to check.
+        self.module_attrs = ['platform', 'name', 'tags', 'inst_ids',
+                             'load', 'list_files', 'download']
+        self.inst_attrs = ['tag', 'inst_id', 'acknowledgements', 'references',
+                           'inst_module']
+        self.inst_callable = ['load', 'list_files', 'download', 'clean',
+                              'default']
+        self.attr_types = {'platform': str, 'name': str, 'tags': dict,
+                           'inst_ids': dict, 'tag': str, 'inst_id': str,
+                           'acknowledgements': str, 'references': str}
```

Comment:
Reminding myself that moving these here will probably break things.  These could be moved back, or moved to a method-level setup.

Reply:
These need to be defined outside the `setup_class` for backwards compatibility.  Comment added.